Add getting started and installation instructions to the welcome page

weiji14 opened the pull request:
Editing the front page to include a one-click Binder link to run the Jupyter notebooks, and steps for people to get a conda virtual environment set up locally for the EGU22 short course.

Once everyone has their Jupyter notebooks on the website at https://www.generic-mapping-tools.org/egu22pygmt, I'm hoping to make a quick 5-ish minute video that will introduce PyGMT very briefly and walk through how people can navigate the Jupyter Book, click on the Binder links, etc.

weiji14 committed:
Add getting started and installation instructions to the welcome page

Editing the front page to include a one-click Binder link to run the Jupyter notebooks, and steps for people to get a conda virtual environment set up locally for the EGU22 short course.

weiji14 committed:
Merge branch 'main' into doc/welcome-page

weiji14 committed:
Revert "Temporarily trigger jupyter book build on CI"

This reverts commit f584d3166a24bb805743a637a58b61b9a7268e4f.

weiji14 committed:
Temporarily trigger jupyter book build on CI

weiji14 committed:
Use list of tutorials and authors instead of tableofcontents

weiji14 commented on book/intro.md:
```diff
@@ -1,11 +1,89 @@
-Check out the content pages bundled with this sample book to see more.
+- [Anatomy of a PyGMT figure](first-figure) by [Leonardo Uieda](https://orcid.org/0000-0001-6123-9515)
+- [Integration with the scientific Python ecosystem 🐍](ecosystem) by [Max Jones](https://orcid.org/0000-0003-0180-8928)
+- [Making some Mars maps with pygmt](Mars_Maps) by [André Luiz Belém](https://orcid.org/0000-0002-8865-6180)
```
Placeholder for André's tutorial. Will update the link once the notebook is in this repo.

Oh, and do you think it's ok to put it here between the ecosystem and LiDAR tutorials? Or move it to another position?

maxrjones replied:
Yes, I think this is a good spot.

weiji14 commented on .github/workflows/deploy-book.yml:
```diff
@@ -4,7 +4,7 @@ name: deploy-book
 on:
   # Uncomment the 'pull_request' line below to manually re-build Jupyter Book
-  # pull_request:
+  pull_request:
```
TODO comment this out before merging. Just having this run in the PR so things are rendered at https://www.generic-mapping-tools.org/egu22pygmt
```suggestion
  # pull_request:
```

weiji14 replied:
Done in 5e0006b037b9ce9e166788632fdd6b4e6db6dd1a

weiji14 commented on book/intro.md:
```diff
@@ -1,11 +1,89 @@
 
-Check out the content pages bundled with this sample book to see more.
+- [Anatomy of a PyGMT figure](first-figure) by [Leonardo Uieda](https://orcid.org/0000-0001-6123-9515)
+- [Integration with the scientific Python ecosystem 🐍](ecosystem) by [Max Jones](https://orcid.org/0000-0003-0180-8928)
```
Max, let me know if you prefer to use your new name here. I can also update your name on the main README.md. There should be a way to change your name at https://administrator.copernicus.org/personal_data, though I'm not sure if that will show up immediately on the abstract page.

maxrjones replied:
Thanks for checking in. This is good here and it would be nice to update the readme (let me know if you want me to submit a PR). I just updated my EGU profile.

weiji14 replied:
Sure, you can start a PR to update the main README.md. Maybe a good idea to update the 'Schedule' table too.

weiji14 committed:
Temporary commit to show Mars Maps tutorial on sidebar

TODO revert this commit once André's notebook is finished.

weiji14 committed:
Merge branch 'main' into doc/welcome-page

weiji14 committed:
Use sphinx-panels with thumbnails of tutorials

Change from using bulleted list of tutorials to a more eye-catching 4-grid-panel with a teaser image for each tutorial.

weiji14 commented on book/intro.md:
````diff
@@ -1,11 +1,121 @@
+---
+:img-top: https://user-images.githubusercontent.com/23487320/168939109-e35c9d85-4b2d-4623-abd8-66ed1148cf21.png
+```{link-button} Mars_Maps
+:text: Making some Mars maps with PyGMT
````
TODO update the thumbnail image to point to the rendered version once Andre's jupyter notebook is uploaded to the repo, see #15

weiji14 committed:
Fix thumbnail image links

weiji14 committed:
Remove bold heading from LiDAR tutorial

Make the font consistent with the other tutorials.

weiji14 reviewed (commented):
Managed to make the front-page nicer using sphinx-panels (see https://jupyterbook.org/en/stable/content/content-blocks.html#panels). Now it looks like this:

![image](https://user-images.githubusercontent.com/23487320/168949017-e32e954a-a614-42b6-b862-21cdd00fcf11.png)

Feel free to give this PR a review. I'll start to record the 5-10min intro video using this front-page now to get a feel, but will probably finish it by tomorrow morning.

weiji14 committed:
Fix incorrect link

weiji14 commented:
Ok, first shot at the intro, just taking people through an outline of the course content and teaching them how to get set up on Binder or locally.

https://user-images.githubusercontent.com/23487320/168953934-e58a2627-06b1-46ed-9a9c-0ce2f1d4b832.mp4

Will make another recording tomorrow morning at uni (with better lighting and a proper background) after a bit more practice. Oh, and remind myself to bump up the font size too.

leouieda commented on book/intro.md:
```diff
@@ -1,11 +1,121 @@
-# Welcome to your Jupyter Book
+# 🥳 Welcome to the EGU22 PyGMT short course
```
```suggestion
# Crafting beautiful maps with PyGMT

## 🥳 Welcome to the EGU22 PyGMT short course
```

The h1 in this page is picked up as the title for the Jupyter book, which maybe should be the title of the course? Then we can have the welcome message as an h2 beneath it.

weiji14 replied:
Sounds good, done in f0592acf249d3543f41222b84cb93c5e646ae21a

leouieda reviewed (approved):
@weiji14 front page looks great and the intro video as well. Left a small suggestion (feel free to ignore).

yvonnefroehlich commented:
@weiji14 First of all the front page looks already very nice!

However there is a small error in the installation instruction of the `egu22pygmt` environment at step 4 in the second line:
The `env` is missing. For me works:
`conda`**`env`**`create --name egu22pygmt --file environment.yml`

Without the `env` I get the error:
CondaValueError: could not parse 'name: egu22pygmt' in: environment.yml

weiji14 committed:
Modify heading to be title of workshop

Co-Authored-By: Leonardo Uieda <leouieda@gmail.com>

weiji14 committed:
Fix conda virtual environment creation command

Co-Authored-By: Yvonne Fröhlich <94163266+yvonnefroehlich@users.noreply.github.com>

weiji14 commented:
Glad to see that people are trying out the short course already :laughing: Thanks @yvonnefroehlich, I've fixed the command in 05fd56819cd1ccc26c01a00a2d88d0906493575b.

weiji14 committed:
Put in more emojis on the front sections

weiji14 commented on book/intro.md:
```diff
@@ -1,11 +1,123 @@
+🖱️ click on the button below to launch on regular
+[Binder](https://mybinder.readthedocs.io/en/latest/index.html).
+
+[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/GenericMappingTools/egu22pygmt/main)
```
Note to self, add another Binder and Google Colab option as backups in case too many people try to use this at once during EGU. From memory, I think there's a 50 or 100 people limit on how many BInder instances can be launched per GitHub repo.

weiji14 replied:
Might need to increase limits, am getting `toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit`. Form to request additional resouces for MyBinder: https://github.com/jupyterhub/mybinder.org-deploy/issues/new?assignees=&labels=impact&template=request_resources.md. Looking at a few issues, they may be stretched on funding recently...

weiji14 replied:
Have added a backup Pangeo Binder link in 10593904d87fd4c5ff4a5c9a1b7f85de022f9d7c. Haven't got the bandwidth to do Google Colab but maybe someone can work on it in a separate PR.

maxrjones commented on book/intro.md:
````diff
@@ -1,11 +1,123 @@
+:text: Making some Mars maps with PyGMT
+:type: ref
+```
+by [André Luiz Belém](https://orcid.org/0000-0002-8865-6180)
````
```suggestion
by [André Belém](https://orcid.org/0000-0002-8865-6180)
```
suggestion based on https://github.com/GenericMappingTools/egu22pygmt/pull/16/files

weiji14 committed:
Remove André's middle name

Co-authored-by: Max Jones <meghanj@alum.mit.edu>

weiji14 committed:
Temporarily use gke.mybinder.org

weiji14 commented:
For completeness, here's the final recording of the intro video :tada:

https://user-images.githubusercontent.com/23487320/169176197-06f30c97-e926-44e8-be89-6ae703a9c608.mp4

Have already stitched the 1 intro + 4 tutorial videos together, and it just fits under 1hr 30min (let's pretend I didn't have to re-record the intro too many times). Tried doing 1080p60fps but that produced a 7GB file so no go. Did 1080p30fps and that came down to 4GB. Decided to try uploading that 4GB version but the upload seemed to stop halfway (wifi at home not strong enough). I'm doing 1080p23.98fps now and seeing if that can cut the size. Few more hours to go :relaxed:

maxrjones commented:
Do you want me to see what I get with handbrake? That's the tool I've used for processing community meeting videos

weiji14 commented:
> Do you want me to see what I get with handbrake? That's the tool I've used for processing community meeting videos

Too late, I've emailed the video in already (1080p23.8fps). It's 3.3GB though, let's just hope they're lenient enough.

maxrjones commented:
> Too late, I've emailed the video in already (1080p23.8fps). It's 3.3GB though, let's just hope they're lenient enough.

Sounds good, thanks for processing the video!

Also, great job on the intro! it's fantastic 🎉

leouieda commented:
@weiji14 thank you for all the work you've done on this! The intro does look really good. 

We can upload higher resolution videos to YouTube later since EGU had this silly restriction.

weiji14 committed:
Merge branch 'main' into doc/welcome-page

weiji14 committed:
Use relative link to Mars Maps image

weiji14 committed:
Remove Mars_Maps.md and revert gke.mybinder.org workaround

maxrjones commented on book/Mars_Maps.md:
```diff
@@ -0,0 +1,3 @@
+# Making some Mars maps with PyGMT
+
+TODO
```
Just a reminder to remove this file when you have time

weiji14 replied:
Yep, done in 414c38c953e15e7e2ed4cf74a0e735141f3bcb1b!

weiji14 committed:
Add note about using Pangeo Binder as a backup to regular mybinder

maxrjones reviewed (approved):
Looks so great! Thanks for leading this effort!
